packages/wallet-apis/src/types.ts

```diff
@@ -34,6 +34,10 @@ export type OptionalChainId<T> = T extends { chainId: Hex }
   ? Omit<T, "chainId"> & { chainId?: Hex | undefined }
   : T;
 
+export type OptionalSignerAddress<T> = T extends { signerAddress: Address }
```

Comment:
ultra small nit but is there a reason we need to check that T extends { signerAddress: Address } ?

I ask cause I feel like using `OptionalSignerAddress` on a type that doesn't have a signer address is a mistake and the compiler should flag it